http.go

```diff
@@ -49,7 +50,11 @@ func SendHTTPRequest(url, endpoint, method, query, body string, headers map[stri
 		req.Header.Add(key, value)
 	}
 	logger.Debug("sending HTTP request: %+v", req)
-	resp, err := http.DefaultClient.Do(req)
+	// this makes a custom http client with a timeout of 10 secs for each request
+	var httpClient = &http.Client{
+		Timeout: time.Second * 10,
```

Comment:
as i can see the only think that i could do to make this more `every problem safe` is to receive the timeout as a parameter